### PR TITLE
fix(sign-up): add module siret for corporation fr

### PR DESCRIPTION
**Pull request by ghyenne**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop` 
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | MANAGER-8600 
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description
I added a new module siret for corporation fr

## Related
:flags: Translations: 
- [x] TRANS-47789

### CODE REVIEW

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            class="mb-3"
+        ></span>
+
+        <oui-field data-ng-if="$ctrl.suggest.entryList.length > 0">
+            <oui-select-picker
+                data-ng-repeat="suggest in $ctrl.suggest.entryList"
+                name="siretPicker"
+                label="{{ suggest.name }}"
+                on-change="$ctrl.selectSuggest(modelValue)"
+                values="[suggest]"
```
Comment (anooparveti) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:69]:
```suggestion
                data-on-change="$ctrl.selectSuggest(modelValue)"
                data-values="[suggest]"
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            data-translate="{{ $ctrl.suggest.entryList.length === 0 ? 'siret_suggest_no_result' : 'siret_suggest_select_company' }}"
+            data-translate-values="{ search: '<strong>'+ $ctrl.search +'</strong>'}"
+            class="mb-3"
+        ></span>
+
+        <oui-field data-ng-if="$ctrl.suggest.entryList.length > 0">
+            <oui-select-picker
+                data-ng-repeat="suggest in $ctrl.suggest.entryList"
+                name="siretPicker"
+                label="{{ suggest.name }}"
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:67]:
```suggestion
                data-name="siretPicker"
                data-label="{{ suggest.name }}"
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+                on-change="$ctrl.selectSuggest(modelValue)"
+                values="[suggest]"
+                required
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:70]:
```suggestion
                data-required
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+        <div
+            class="mt-1 mb-3"
+            data-ng-if="!$ctrl.model.companyNationalIdentificationNumber"
+        >
+            <span
+                class="align-middle"
+                data-translate="siret_suggest_no_found_company"
+            ></span>
+            <button
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:98]:
```suggestion
            <button
                type="button"
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+    <!-- end suggest form -->
+
+    <!-- manual form -->
+    <div data-ng-if="$ctrl.displayManualForm">
+        <span data-translate="siret_manual_title"></span>
+
+        <div class="row mt-2">
+            <oui-field
+                class="col-lg-6"
+                label="{{ 'siret_manual_company_name' | translate }}"
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:132]:
```suggestion
                data-label="{{ 'siret_manual_company_name' | translate }}"
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+            <oui-field
+                class="col-lg-6"
+                label="{{ 'siret_manual_company_legal_form' | translate }}"
+            >
+                <oui-select
+                    items="$ctrl.legalFormList"
+                    name="legalFormField"
+                    model="$ctrl.model.corporationType"
+                    required
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:152]:
```suggestion
                    data-required
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+                    name="siretField"
+                    data-ng-model="$ctrl.model.companyNationalIdentificationNumber"
+                    type="text"
+                />
+            </oui-field>
+
+            <oui-field
+                class="col-lg-4"
+                label="{{ 'siret_vat' | translate }} ({{ 'siret_optional' | translate }})"
+                label-popover="{{ 'siret_vat_helper' | translate }}"
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:174]:
```suggestion
                data-label="{{ 'siret_vat' | translate }} ({{ 'siret_optional' | translate }})"
                data-label-popover="{{ 'siret_vat_helper' | translate }}"
```

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        </div>
+    </div>
+    <!-- end manual form -->
+
+    <!-- valid that the information is correct -->
+    <oui-checkbox
+        class="mb-4"
+        data-ng-if="$ctrl.suggest.type === 'siret' || $ctrl.displayManualForm"
+        name="isCorrectCheckbox"
+        required
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.html:204]:
```suggestion
        data-name="isCorrectCheckbox"
        data-required
```

File: packages/manager/modules/sign-up/src/components/siret/siret.service.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    this.$locale = $locale;
+  }
+
+  getSiret(params) {
+    return this.$http
+      .get('/me/suggest/company', { params })
+      .then(({ data }) => data)
+      .catch(() => ({
+        entryList: [],
+      }));
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/siret.service.js:14]: **question**: doing so prevents any errors from being catched on this promise. Is this what we want ?

File: packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json [hunk 10]
```diff
@@ -0,0 +1,32 @@
+{
+  "siret_search_intro": "Rechercher votre entreprise par :",
```
Comment (y4nnL) [inline packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json:2]:
```suggestion
  "siret_search_intro": "Recherchez votre entreprise par :",
```

File: packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  "siret_search_field_placeholder": "N° SIRET, N° SIREN ou nom de l'entreprise",
+  "siret_search_button": "Chercher",
+  "siret_suggest_select_company": "Sélectionnez votre entreprise",
+  "siret_suggest_many_result_title": "Votre recherche comporte un grand nombre de résultats.",
+  "siret_suggest_many_result_description": "Si vous ne trouvez pas votre entreprise, veuillez préciser votre recherche ou saisir votre numéro SIRET",
+  "siret_suggest_no_result": "Nous n'avons pas trouvé \"{{search}}\"",
+  "siret_suggest_no_found_company": "Vous ne trouvez pas votre entreprise ?",
+  "siret_suggest_add_manual": "Ajoutez-la manuellement",
+  "siret_suggest_siret": "SIRET :",
+  "siret_suggest_siren": "SIREN :",
```
[the remark this answers is not in the document]
Comment (y4nnL) [reply packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json:13]: Those are registration numbers of French companies, IMO it should be translated as it make no sense for other languages

**Review by y4nnL — CHANGES_REQUESTED** [review]

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 1 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 12]
```diff
@@ -0,0 +1,210 @@
...
+            >
+                <oui-select-picker-section>
+                    <span
+                        data-ng-if="$ctrl.suggest.type !== 'name'"
+                        data-ng-bind="suggest.address + ' ' + suggest.city + ' ' + suggest.country"
+                    ></span>
+                    <span
+                        class="d-block"
+                        data-ng-bind="('siret_suggest_siren' | translate ) +' '+ suggest.primaryCNIN"
+                    ></span>
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied. With <p> and class="mb-0"

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 13]
```diff
@@ -0,0 +1,210 @@
...
+                    <span
+                        class="d-block"
+                        data-ng-if="$ctrl.suggest.type !== 'name'"
+                        data-ng-bind="('siret_suggest_siret' | translate ) +' '+ suggest.secondaryCNIN"
+                    ></span>
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied. With <p> class="mb-0"

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 14]
```diff
@@ -0,0 +1,210 @@
...
+                    class="oui-icon oui-icon-arrow-right"
+                    aria-hidden="true"
+                ></span>
+            </button>
+        </div>
+
+        <oui-field
+            data-ng-if="$ctrl.suggest.type === 'siret'"
+            label="{{ 'siret_vat' | translate }} ({{'siret_optional' | translate }})"
+            label-popover="{{ 'siret_vat_helper' | translate }}"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+            </oui-field>
+
+            <oui-field
+                class="col-lg-6"
+                label="{{ 'siret_manual_company_legal_form' | translate }}"
+            >
+                <oui-select
+                    items="$ctrl.legalFormList"
+                    name="legalFormField"
+                    model="$ctrl.model.corporationType"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/apps/sign-up/src/form/form.controller.js [hunk 16]
```diff
@@ -38,6 +38,15 @@ export default class SignUpFormAppCtrl {
     }
   }
 
+  onStepperCancelled(step) {
+    this.atInternet.trackPage({
+      name: `accountcreation-step${step === 'details' ? '2' : '3'}-${
+        this.me.model.legalform
+      }::cancel`,
+    });
+    return this.onStepCancel(step);
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/apps/sign-up/src/form/form.controller.js]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/index.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+angular
+  .module(moduleName, [
+    'ngAtInternet',
+    'ngTranslateAsyncLoader',
+    'oui',
+    'pascalprecht.translate',
+  ])
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .component(component.name, component)
+  .service('siretService', service);
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/index.js]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.component.js [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import template from './siret.html';
+import controller from './siret.controller';
+
+export default {
+  name: 'siretComponent',
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.component.js]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 2 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 19]
```diff
@@ -0,0 +1,210 @@
...
+            </oui-field>
+
+            <oui-field
+                class="col-lg-5"
+                label="{{ 'siret_manual_registration_date' | translate }} ({{ 'siret_optional' | translate }})"
+            >
+                <!-- for now this value isn't save, this will come later -->
+                <oui-calendar
+                    name="registrationDateField"
+                    alt-format="{{ $ctrl.siretService.dateFormat() }}"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 3 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 4 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html:99]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 5 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 6 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 20]
```diff
@@ -0,0 +1,210 @@
...
+                    required
+                >
+                </oui-select>
+            </oui-field>
+        </div>
+
+        <div class="row">
+            <oui-field
+                class="col-lg-3"
+                label="{{ 'siret_siret' | translate }} ({{ 'siret_optional' | translate }})"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html (hunk 7 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/siret.html [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+                    class="oui-input"
+                    name="vatField"
+                    data-ng-model="$ctrl.model.vat"
+                    type="text"
+                />
+            </oui-field>
+
+            <oui-field
+                class="col-lg-5"
+                label="{{ 'siret_manual_registration_date' | translate }} ({{ 'siret_optional' | translate }})"
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.html]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json (hunk 10 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/index.js [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+
+angular
+  .module(moduleName, [
+    'ngAtInternet',
+    'ngTranslateAsyncLoader',
+    'oui',
+    'pascalprecht.translate',
+  ])
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .component(component.name, component)
```
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/index.js]: It's applied.

File: packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json (hunk 11 above)
[the remark this answers is not in the document]
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/translations/Messages_fr_FR.json:13]: In fact it is content, even if "SIRET" is not translated, the space between "SIRET" and ":" is present or not depending on the language. I don't know if you can put another language when you put country of residence FR.

File: packages/manager/modules/sign-up/src/components/siret/siret.service.js (hunk 9 above)
Comment (ghyenne) [reply packages/manager/modules/sign-up/src/components/siret/siret.service.js:14]: Currently the errors are not really interpretable and could not be translated.
The best thing to do is not to have any results and to propose to the user to switch to manual mode.

**Review by anooparveti — DISMISSED** [review]

**Review by y4nnL — DISMISSED** [review]

**Review by sachinrameshn — APPROVED** [review]

**Pull request merged**